soh/soh/Enhancements/timesaver_hook_handlers.cpp

```diff
@@ -769,6 +781,61 @@ void TimeSaverOnActorInitHandler(void* actorRef) {
         });
     }
 
+    if (actor->id == ACTOR_BG_SPOT03_TAKI) {
+        bgSpot03UpdateHook = GameInteractor::Instance->RegisterGameHook<GameInteractor::OnActorUpdate>([](void* innerActorRef) mutable {
+            Actor* innerActor = static_cast<Actor*>(innerActorRef);
+
+            if (innerActor->id != ACTOR_BG_SPOT03_TAKI) {
+                return;
+            }
+
+            bool shouldKeepOpen;
+            switch (CVarGetInteger(CVAR_ENHANCEMENT("TimeSavers.SleepingWaterfall"), 0)) {
+                case 1:
+                    shouldKeepOpen = Flags_GetEventChkInf(EVENTCHKINF_OPENED_ZORAS_DOMAIN);
+                    break;
+                case 2:
+                    if (IS_RANDO) {
+                        shouldKeepOpen = RAND_GET_OPTION(RSK_SLEEPING_WATERFALL) == RO_WATERFALL_OPEN;
```

Comment:
@Malkierian This is the cause of Never not working when the waterfall is closed. The IS_RANDO check should fall back to the inventory check if the setting is closed if the intent is for the enhancement is to always apply to closed waterfall, but it won't as that's in an else block.

Reply:
Yeah, I noticed that too. I'll fix it, it's easy.